### PR TITLE
docs : add documentation for hook functions #67

### LINKED ISSUE #67: Complete documentation for functions for hooks files
- This issue has no `code change`
- You only have to write `documentations`
- `Documentations` should have the standard of [JsDoc](https://jsdoc.app/)
- Don't add `param types` and `return types`, as we are using typescript
- Only Add documentation for `useAuth.ts`, `useAuthHeader.ts`, `useAuthUser.ts`, `useIsAuthenticated.ts`, `useSignIn.ts` and `useSignOut.ts`

Please read the [CONTRIBUTING.md](https://github.com/react-auth-kit/react-auth-kit/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by Hardik0307**
- [x] I have read the **CONTRIBUTING** document.
- [x] added document for all hook file functions.
- [x] followed [JsDoc](https://jsdoc.app/)

Kindly Review.

Fixes #67

### CODE REVIEW

**Comment by Hardik0307** [comment]
@darkmatter18  check test considering `no-trailing-spaces` for comments too.

to avoid we have to do,
in the, **package.json**
> npm test:lint :  eslint ...  --fix

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "test": "run-s test:unit test:lint test:build",
     "test:build": "run-s build",
     "test:unit": "jest",
-    "test:lint": "eslint src/**/*.ts src/**/*.tsx",
+    "test:lint": "eslint src/**/*.ts src/**/*.tsx --fix",
```
Comment (darkmatter18) [inline package.json:25]: Fix Command is there, please undo it

**Review by darkmatter18 — CHANGES_REQUESTED** [review]
Nice Work. But just a minor change and we are good to go.

File: package.json (hunk 1 above)
Comment (Hardik0307) [reply package.json]: Resolved @darkmatter18 .
Have a look.

**Comment by darkmatter18** [comment]
LGTM. Thanks for your contribution. @Hardik0307 🥇

**Pull request merged**